cpp/src/arrow/flight/sql/server.h

```diff
@@ -686,50 +686,50 @@ class ARROW_FLIGHT_SQL_EXPORT SqlSchema {
  public:
   /// \brief Get the Schema used on GetCatalogs response.
   /// \return The default schema template.
-  static std::shared_ptr<Schema> GetCatalogsSchema();
+  static const std::shared_ptr<Schema>& GetCatalogsSchema();
```

Comment:
Why we are changing these functions to return const references?
I think that won't increase the reference counter and can cause potential problems.

Reply:
You can always copy the pointer if you want to…but the original signature forced you to take a copy which isn't always desirable

Reply:
In which cases it was not ideal? when invoking from `FlightSqlServerBase::DoGet`?
If that's the case, I think there are other `DoGet*Schema`s that keep returning just a shared ptr instead a reference.

Reply:
Then ideally we should make them return a reference